tutorials/cloud-spanner-aspnet/index.md

```diff
@@ -0,0 +1,107 @@
+
+![image alt text](image_3.png)
+
+### **Getting Books.**
+
+Now that we have code to insert records, we can add more code to list all books or a single book.
+
+Add the following methods to your controller.
+
+    . . .
+    [HttpGet]
    public async Task<IActionResult> GetAll()
    {
        var result = new List<Book>();
     
        using (var connection = new SpannerConnection(
            $"Data Source=projects/{_myProject}/instances/myspanner/databases/books"))
        {
            var selectCmd = connection.CreateSelectCommand("SELECT * FROM bookTable");
            using (var reader = await selectCmd.ExecuteReaderAsync())
            {
                while (await reader.ReadAsync())
                {
                    result.Add(new Book
                    {
                        Id = reader.GetFieldValue<string>("ID"),
                        Title = reader.GetFieldValue<string>("Title"),
                        Author = reader.GetFieldValue<string>("Author"),
                        PublishDate = reader.GetFieldValue<DateTime>("PublishDate")
                    });
                }
            }
        }
     
        return Ok(result);
    }
     
    [HttpGet]
    [Route("{id}", Name="GetBookById")]
    public async Task<IActionResult> Get(string id)
    {
        using (var connection = new SpannerConnection(
            $"Data Source=projects/{_myProject}/instances/myspanner/databases/books"))
        {
            var selectCommand =connection.CreateSelectCommand($"SELECT * FROM bookTable WHERE ID='{id}'");
            using (var reader = await selectCommand.ExecuteReaderAsync())
            {
                while (await reader.ReadAsync())
                {
                    return Ok(new Book
                    {
                        Id = reader.GetFieldValue<string>("ID"),
                        Title = reader.GetFieldValue<string>("Title"),
                        Author = reader.GetFieldValue<string>("Author"),
                        PublishDate = reader.GetFieldValue<DateTime>("PublishDate")
                    });
                }
            }
        }
     
        return NotFound();
    }
+    . . .
+
+
+Now, run your program.  Using Postman, perform an HTTP GET on [http://localhost:56777/api/books](http://localhost:56777/api/books) to return all books or [http://localhost:56777/api/books/[ID](http://localhost:56777/api/books/[ID)] to retrieve a single book.
+
+### **[Optional] Deploy to Google App Engine Flex**
```

Comment:
Fix path formatting and branding here. 

Suggest: 
## (Optional) Deploying the application to App Engine 

For automatic scaling and management, you can deploy your Web API project to the App Engine flexible environment.

1.  Download and install [Google Cloud Tools for Visual Studio](https://cloud.google.com/tools/visual-studio/docs/quickstart#install_cloud_tools_for_visual_studio). 
1.  In the main menu, select** Tools > Google Cloud > Publish to Google Cloud**.
1.  In the Publish dialog box, choose **App Engine Flex**.
1.  Click **Publish**. After publishing completes, the application is live at <code>https://YOUR_PROJECT_ID.appspot.com</code>.